warehouse/integrations/deltalake-native/deltalake_native_test.go

```diff
@@ -0,0 +1,388 @@
+package deltalake_native_test
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/rudderlabs/rudder-server/warehouse/integrations/deltalake"
+
+	deltalake_native "github.com/rudderlabs/rudder-server/warehouse/integrations/deltalake-native"
+
+	"github.com/rudderlabs/rudder-server/warehouse/encoding"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-server/warehouse/validations"
+
+	"github.com/rudderlabs/rudder-server/utils/misc"
+
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+
+	warehouseclient "github.com/rudderlabs/rudder-server/warehouse/client"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-server/warehouse/integrations/testhelper"
+)
+
+func TestIntegrationDeltalake(t *testing.T) {
+	if os.Getenv("SLOW") == "0" {
+		t.Skip("Skipping tests. Remove 'SLOW=0' env var to run them.")
+	}
+	if _, exists := os.LookupEnv(testhelper.DeltalakeIntegrationTestCredentials); !exists {
+		t.Skipf("Skipping %s as %s is not set", t.Name(), testhelper.DeltalakeIntegrationTestCredentials)
+	}
+
+	t.Parallel()
+
+	deltalake.Init()
+	deltalake_native.Init()
+
+	credentials, err := testhelper.DatabricksCredentials()
+	require.NoError(t, err)
+
+	nativeCredentials := deltalake_native.Credentials{
+		Host:  credentials.Host,
+		Port:  credentials.Port,
+		Path:  credentials.Path,
+		Token: credentials.Token,
+	}
+
+	dl := deltalake_native.NewDeltalake()
+	deltalake_native.WithConfig(dl, config.Default)
+
+	db, err := deltalake_native.Connect(nativeCredentials)
+	require.NoError(t, err)
+
+	var (
+		jobsDB       = testhelper.SetUpJobsDB(t)
+		provider     = warehouseutils.DELTALAKE
+		schema       = testhelper.Schema(provider, testhelper.DeltalakeIntegrationTestSchema)
+		nativeSchema = fmt.Sprintf("%s_%s", schema, "native")
+	)
+
+	testCases := []struct {
+		name          string
+		writeKey      string
+		sourceID      string
+		destinationID string
+		schema        string
+		useLegacy     bool
+	}{
+		{
+			name:          "Native",
+			writeKey:      "dasFgoilA0U1WxNeW1gdgUVDfas",
+			sourceID:      "36H5EpYzojqQSepRSaGBrrPx3e4",
+			destinationID: "36IDjdnoEus6DDNrth3SWO1FOpu",
+			schema:        nativeSchema,
+			useLegacy:     false,
+		},
+		{
+			name:          "Legacy",
+			writeKey:      "sToFgoilA0U1WxNeW1gdgUVDsEW",
+			sourceID:      "25H5EpYzojqQSepRSaGBrrPx3e4",
+			destinationID: "25IDjdnoEus6DDNrth3SWO1FOpu",
+			schema:        schema,
+			useLegacy:     true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Cleanup(func() {
+			require.NoError(
+				t,
+				testhelper.WithConstantBackoff(func() (err error) {
+					_, err = db.Exec(fmt.Sprintf(`DROP SCHEMA %[1]s CASCADE;`, tc.schema))
+					return
+				}),
+				fmt.Sprintf("Failed dropping schema %s for Deltalake", tc.schema),
```

Comment:
If you were to use `require.Eventually` instead of `require.NoError`+`WithConstantBackoff` it would be clearer to the reader to how long you're going to try for and how often. Plus you'd be using one function only.